app/page.tsx

```diff
@@ -90,25 +82,49 @@ const HomeSearchPage: React.FC = () => {
     availableOnly: DEFAULT_AVAILABILITY_FILTER_OPTIONS,
     price_min: DEFAULT_MIN_PRICE,
     price_max: DEFAULT_MAX_PRICE,
-  });
+  };
+
+  const [searchParams, setSearchParams] = useState<searchParamsType | null>(null);
```

Comment:
Why not initially set it to the default.

Reply:
```suggestion
  const [searchParams, setSearchParams] = useState<searchParamsType>(defaultSearchParams);
```
This avoids the hassle of making the type nullable and subsequently dealing with it whenever we want to use `searchParams`.

Reply:
I would like to remove this, but this breaks `Sidebar.tsx` and resets to default values on refresh. Out of sync with searchParams in page.tsx and localStorage.

We should focus on other features like UI/UX for now. Maybe, we can revisit this later.